ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -169,202 +176,316 @@ public static KsqlEngine create(final KsqlConfig ksqlConfig) {
     );
   }
 
-  static KsqlEngineMetrics createKsqlEngineMetrics(final KsqlEngine ksqlEngine) {
-    return new KsqlEngineMetrics("ksql-engine", ksqlEngine);
+  /**
+   * Parse the statements, but do NOT update the metastore.
+   *
+   * @param sql the statements to parse
+   * @return the list of prepared statements.
+   */
+  public List<PreparedStatement<?>> parseStatements(final String sql) {
+    try {
+      final MetaStore parserMetaStore = metaStore.clone();
+
+      final KsqlParser ksqlParser = new KsqlParser();
+
+      return ksqlParser.buildAst(
+          sql,
+          parserMetaStore,
+          stmt -> {
+            validateSingleQueryAstAndUpdateParserMetaStore(stmt, parserMetaStore);
+          });
+    } catch (final KsqlException e) {
+      throw e;
+    } catch (final Exception e) {
+      throw new KsqlStatementException(
+          "Exception while processing statements: " + e.getMessage(), sql, e);
+    }
   }
 
   /**
-   * Builds the set of queries in the given query string, updating the metastore.
+   * Try to execute the supplied statements. No internal state will be changed.
    *
-   * <p>The queries are not started.
+   * <p>Statements must be executable. See {@link #isExecutableStatement(PreparedStatement)}.
    *
-   * @param queriesString The ksql query string.
+   * <p>No changes will be made to the meta store and no queries will be registered.
+   *
+   * <p>If the statements contains queries, they will not be added to the active set of queries and
+   * they will have been closed before returning.
+   *
+   * @param statements the list of statements to execute
+   * @param ksqlConfig the base ksqlConfig
+   * @param overriddenProperties the property overrides
+   * @return List of query metadata.
+   */
+  public List<QueryMetadata> tryExecute(
+      final List<? extends PreparedStatement<?>> statements,
```

Comment:
This should really be Query, as it was before. We don't want to "try-execute" CSAS/CTAS statements, because they can have side-effects. For example, I believe we will now wind up creating sink topics when validating statements from the rest API. Longer term, we should consider passing in a dependency that abstracts away all the possible side-effects (topic creation, id generation, etc), with a mock/validating implementation for try-execute. But for this PR I think we should switch this back to just building bare queries..